tracer/src/Datadog.Trace/OTelMetrics/OtlpMetricsSerializer.cs

```diff
@@ -0,0 +1,573 @@
+
+            WriteStringField(writer, FieldNumbers.Name, meterName);
+            WriteStringField(writer, FieldNumbers.Version, string.Empty);
+
+            return buffer.ToArray();
+        }
+
+        private byte[] SerializeCounterMetric(MetricPoint metric)
+        {
+            using var buffer = new MemoryStream();
+            using var writer = new BinaryWriter(buffer, Encoding.UTF8);
+
+            WriteStringField(writer, FieldNumbers.MetricName, metric.InstrumentName);
+            WriteStringField(writer, FieldNumbers.Description, metric.Description);
+            WriteStringField(writer, FieldNumbers.Unit, metric.Unit);
+
+            // Sum
+            var sumData = SerializeSum(metric);
+            WriteTag(writer, FieldNumbers.Sum, LengthDelimited);
+            WriteVarInt(writer, sumData.Length);
+            writer.Write(sumData);
+
+            return buffer.ToArray();
+        }
+
+        private byte[] SerializeGaugeMetric(MetricPoint metric)
+        {
+            using var buffer = new MemoryStream();
+            using var writer = new BinaryWriter(buffer, Encoding.UTF8);
+
+            WriteStringField(writer, FieldNumbers.MetricName, metric.InstrumentName);
+            WriteStringField(writer, FieldNumbers.Description, metric.Description);
+            WriteStringField(writer, FieldNumbers.Unit, metric.Unit);
+
+            var gaugeData = SerializeGauge(metric);
+            WriteTag(writer, FieldNumbers.Gauge, LengthDelimited);
+            WriteVarInt(writer, gaugeData.Length);
+            writer.Write(gaugeData);
+
+            return buffer.ToArray();
+        }
+
+        private byte[] SerializeHistogramMetric(MetricPoint metric)
+        {
+            using var buffer = new MemoryStream();
+            using var writer = new BinaryWriter(buffer, Encoding.UTF8);
+
+            WriteStringField(writer, FieldNumbers.MetricName, metric.InstrumentName);
+            WriteStringField(writer, FieldNumbers.Description, metric.Description);
+            WriteStringField(writer, FieldNumbers.Unit, metric.Unit);
+
+            var histogramData = SerializeHistogram(metric);
+            WriteTag(writer, FieldNumbers.Histogram, LengthDelimited);
+            WriteVarInt(writer, histogramData.Length);
+            writer.Write(histogramData);
+
+            return buffer.ToArray();
+        }
+
+        private byte[] SerializeSum(MetricPoint metric)
+        {
+            using var buffer = new MemoryStream();
+            using var writer = new BinaryWriter(buffer, Encoding.UTF8);
+
+            var dataPointData = SerializeNumberDataPoint(metric);
+            WriteTag(writer, FieldNumbers.DataPoints, LengthDelimited);
+            WriteVarInt(writer, dataPointData.Length);
+            writer.Write(dataPointData);
+
+            int temporality;
+            bool isMonotonic;
+
+            if (metric.InstrumentName.Contains("upDownCounter"))
```

Comment:
nit: I suggest using the instrumentType here since you can easily assert that it's one of two values